packages/cosmos-tigergraph/src/index.ts

```diff
@@ -0,0 +1,185 @@
+      return this.token
+    })
+  }
+
+  public async getTigerGraphData (vertexType: Array<string>, edgeType: Array<string>): Promise<{ nodes: N[]; links: L[] }> {
+    return fetch(`${this.host}:14240/gsqlserver/interpreted_query`, {
+      method: 'POST',
+      body: `INTERPRET QUERY () FOR GRAPH ${this.graphname} {
+          ListAccum<EDGE> @@edges;
+          Seed = {${vertexType.join('.*, ')}.*};
+          Res = SELECT d FROM Seed:d - ((${edgeType.join(' | ')}):e) -> :t
+                  ACCUM @@edges += e;
+          PRINT Seed;
+          PRINT @@edges AS edges;
+        }`,
+      headers: {
+        'Content-Type': 'application/json',
+        Authorization: `Basic ${btoa(`${this.username}:${this.password}`)}`,
+      },
+    }).then(response => {
+      if (!response.ok) {
+        throw new Error(`Error! status: ${response.status}`)
+      }
+
+      return response.json()
+    }).then(data => {
+      const links: L[] = []
+      const nodes: N[] = []
+
+      if (data.error) {
+        throw new Error(`Error! status: ${data.message}`)
+      }
+
+      const vertices = data.results[0].Seed
+      const edges = data.results[1].edges
+      for (const vertex in vertices) {
+        nodes.push({
+          id: `${vertices[vertex].v_type}_${vertices[vertex].v_id}`,
+          v_id: `${vertices[vertex].v_id}`,
+          v_type: `${vertices[vertex].v_type}`,
+          ...(vertices[vertex].attributes),
+
+        })
+      }
+      for (const edge in edges) {
+        links.push({
+          source: `${edges[edge].from_type}_${edges[edge].from_id}`,
+          target: `${edges[edge].to_type}_${edges[edge].to_id}`,
+          ...(edges[edge].attributes),
+        })
+      }
+
+      return { nodes, links }
+    })
+  }
+
+  public async runInterpretedQuery (interpretedQuery: string): Promise<{ nodes: N[]; links: L[] }> {
+    return fetch(`${this.host}:14240/gsqlserver/interpreted_query`, {
+      method: 'POST',
+      body: interpretedQuery,
+      headers: {
+        'Content-Type': 'application/json',
+        Authorization: `Basic ${btoa(`${this.username}:${this.password}`)}`,
+      },
+    }).then(response => {
+      if (!response.ok) {
+        throw new Error(`Error! status: ${response.status}`)
+      }
+
+      return response.json()
+    }).then(data => {
+      const links: L[] = []
+      const nodes: N[] = []
+
+      if (data.error) {
+        throw new Error(`Error! status: ${data.message}`)
+      }
+
+      data = data.results
+
+      for (const res in data) {
+        for (const key in data[res]) {
+          const vertices = data[res][key]
+          for (const vertex in vertices) {
+            if (vertices[vertex].v_type === undefined || vertices[vertex].v_id === undefined) break
+            nodes.push({ ...(vertices[vertex].attributes), ...({ id: `${vertices[vertex].v_type}_${vertices[vertex].v_id}`, v_id: `${vertices[vertex].v_id}`, v_type: `${vertices[vertex].v_type}` }) })
```

Comment:
@GenericP3rson Can we break these initialization into multiple lines, they're too long and a bit difficult to read?

Reply:
Updated in commit https://github.com/cosmograph-org/cosmos-integrations/pull/1/commits/83eff4e6b9e951d6c1702e2faaba5c2720fbd480!